addons/account/models/company.py

```diff
@@ -259,6 +259,11 @@ def _compute_invoice_terms_html(self):
             if html:
                 company.invoice_terms_html = html
 
+    def _get_fiscalyear_last_day(self, year):
+        """ Return a last day of the month based on year."""
+        last_day_of_month = calendar.monthrange(year, int(self.fiscalyear_last_month))[1]
+        return last_day_of_month if self.fiscalyear_last_day != last_day_of_month else self.fiscalyear_last_day
```

Comment:
This won't work if you define a fiscalyear_last_day as not the end of the month.
What you want is the last day of the fiscal year. If it's 29/02, you want it to become 28/02 depending on the year. If not, keep the last_day defined by the user